contract/src/processor/initialize_auction.rs

```diff
@@ -149,6 +149,11 @@ pub fn initialize_auction(
         return Err(AuctionContractError::AuctionIdNotAscii.into());
     }
 
+    // Check if encore period is non-negative
```

Comment:
I think since we deserialize into an u64 it will never be negative. Even if we serialize from a 2s complement negative integer it will be a huge, positive u64. So I'd check a sensible upper bound, i.e. the encore period cannot be greater than 1 hour